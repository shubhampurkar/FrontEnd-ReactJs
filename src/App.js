
import './App.css';
import {BrowserRouter as Router, Route, Switch }from 'react-router-dom';
import ListFlightComponent from './components/ListFlightComponent';

import AddFlightComponent from './components/AddFlightComponent';
import UpdateFlightComponent from './components/UpdateFlightComponent';
import ViewFlightComponent from './components/ViewFlightComponent';
import Login from './pages/login'
import Register from './pages/register';
import ViewAllScheduledFlight from './components/ScheduledFlightpage/viewAllScheduledFlight';

import AdminMain from './pages/AdminMain';
import Booking from './pages/booking';
import ScheduledFlight from './components/ScheduledFlightpage/SCheduleFlight'
import ViewScheduleId from './components/ScheduledFlightpage/viewScheduleId';
import UpdateScheduleFlight from './components/ScheduledFlightpage/UpdateScheduleFlight';
import Addbooking from './components/BookingComponent/Addbooking';
import AddPassenger from './components/BookingComponent/AddPassenger';
import MyOrder from './components/BookingComponent/MyOrder';
import ViewBooking from './components/BookingComponent/ViewBooking';
import DeleteBooking from './components/BookingComponent/BookingDelete'
import SerachScheduled from './components/ScheduledFlightpage/SerachScheduled';


function App() {
  return (
   <div>
     <Router>
       
         <div className="container">
            <Switch>       
              <Route path="/" exact component={Login}></Route>
              <Route path="/register" component={Register}></Route>
              <Route path="/adminmain" component={AdminMain}></Route>
              <Route path="/booking/:id" component = {Booking}></Route>             
              <Route path="/flight" exact component = {ListFlightComponent}></Route>
               <Route path="/viewAllFlights" component = {ListFlightComponent}></Route> 
              <Route path="/addFlight" component = {AddFlightComponent}></Route>
              <Route path="/updateFlight/:id" component = {UpdateFlightComponent}></Route>
              <Route path="/viewFlight/:id" component = {ViewFlightComponent}></Route>
              <Route path="/viewSchedule" component = {ViewAllScheduledFlight}></Route>
              <Route path="/Scheduleflight" component = {ScheduledFlight}></Route>           
              <Route path="/addBooking/:id" component = {Addbooking}></Route>
              <Route path="/addPassenger/:bid/:id" component = {AddPassenger}></Route>
              <Route path="/viewScheduleid/:id" component={ViewScheduleId}></Route>
              <Route path="/myOrder/:id" component={MyOrder}></Route>
              <Route path="/deleteBooking/:id" component={DeleteBooking}></Route>
              <Route path="/viewOrder/:id" component={ViewBooking}></Route>
              <Route path="/searchScheduled" component={SerachScheduled}></Route>
              <Route path="/updateScheduleid/:id" component={UpdateScheduleFlight}></Route>      
              
            </Switch>
             </div> 

      </Router> 
   </div>
  );
}

export default App;
