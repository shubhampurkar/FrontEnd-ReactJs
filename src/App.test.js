import React from 'react';
import App from './App';
import {create} from 'react-test-renderer'

describe('App Snapshot Test',()=>{
  test('testing app Component',()=>{
    let tree=create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})

describe('viewAllScheduledFlight Snapshot test',()=>{
  test('Checking Snapshot',()=>{
    let tree=create(<viewAllScheduledFlight/>)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})

describe('viewScheduledId Snapshot test',()=>{
  test('Checking Snapshot',()=>{
    let tree=create(<viewScheduledId/>)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})


describe('register Snapshot test',()=>{
  test('Checking Snapshot',()=>{
    let tree=create(<register/>)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
describe('booking Snapshot test',()=>{
  test('Checking Snapshot',()=>{
    let tree=create(<booking/>)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
describe('login Snapshot test',()=>{
  test('Checking Snapshot',()=>{
    let tree=create(<login/>)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})







