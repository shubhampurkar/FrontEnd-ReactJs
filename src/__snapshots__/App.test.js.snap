// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Test testing app Component 1`] = `
<div>
  <div
    className="container"
  >
    <div>
      <main
        className="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          className="Login-paper-2"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle Login-avatar-3 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
          </div>
          <h1
            className="MuiTypography-root MuiTypography-h5"
          >
            Sign in
          </h1>
          <form
            className="Login-form-4"
            noValidate={true}
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                  minLength="6"
                  pattern="[a-zA-Z0-9]+"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink={false}
                    htmlFor="userName"
                    id="userName-label"
                  >
                    User Name
                    <span
                      aria-hidden={true}
                      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoComplete="userName"
                      autoFocus={true}
                      className="MuiInputBase-input MuiOutlinedInput-input"
                      disabled={false}
                      id="userName"
                      name="userName"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="text"
                    />
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="PrivateNotchedOutline-legendLabelled-8"
                      >
                        <span>
                          User Name
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <span
                  style={
                    Object {
                      "color": "red",
                      "fontSize": "2",
                    }
                  }
                >
                  
                </span>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                  minLength="6"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink={false}
                    htmlFor="password"
                    id="password-label"
                  >
                    Password
                    <span
                      aria-hidden={true}
                      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoComplete="current-password"
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input"
                      disabled={false}
                      id="password"
                      name="password"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="password"
                    />
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="PrivateNotchedOutline-legendLabelled-8"
                      >
                        <span>
                          Password
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <span
                  style={
                    Object {
                      "color": "red",
                      "fontSize": "2",
                    }
                  }
                >
                  
                </span>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  className="MuiFormControl-root Login-formControl-1"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink={false}
                    htmlFor="outlined-age-native-simple"
                  >
                    UserType
                  </label>
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                    onClick={[Function]}
                    required={true}
                  >
                    <select
                      aria-invalid={false}
                      className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                      disabled={false}
                      multiple={false}
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                    >
                      <option
                        value="customer"
                      >
                        customer
                      </option>
                      <option
                        value="admin"
                      >
                        admin
                      </option>
                    </select>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="PrivateNotchedOutline-legendLabelled-8"
                      >
                        <span>
                          userType
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained Login-submit-5 MuiButton-containedPrimary MuiButton-fullWidth"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Sign In
              </span>
            </button>
            <div
              className="MuiGrid-root MuiGrid-container"
            >
              <div
                align="center"
                className="MuiGrid-root MuiGrid-item"
              >
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                  href="/register"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                >
                  Don't have an account? Sign Up
                </a>
              </div>
            </div>
          </form>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`booking Snapshot test Checking Snapshot 1`] = `<booking />`;

exports[`login Snapshot test Checking Snapshot 1`] = `<login />`;

exports[`register Snapshot test Checking Snapshot 1`] = `<register />`;

exports[`viewAllScheduledFlight Snapshot test Checking Snapshot 1`] = `<viewAllScheduledFlight />`;

exports[`viewScheduledId Snapshot test Checking Snapshot 1`] = `<viewScheduledId />`;
